collector/remote_fx.go

```diff
@@ -0,0 +1,472 @@
+		return err
+	}
+	return nil
+}
+
+type perflibRemoteFxNetwork struct {
+	Name                     string
+	BaseTCPRTT               float64 `perflib:"Base TCP RTT"`
+	BaseUDPRTT               float64 `perflib:"Base UDP RTT"`
+	CurrentTCPBandwidth      float64 `perflib:"Current TCP Bandwidth"`
+	CurrentTCPRTT            float64 `perflib:"Current TCP RTT"`
+	CurrentUDPBandwidth      float64 `perflib:"Current UDP Bandwidth"`
+	CurrentUDPRTT            float64 `perflib:"Current UDP RTT"`
+	FECRate                  float64 `perflib:"FEC Rate"`
+	LossRate                 float64 `perflib:"Loss Rate"`
+	RetransmissionRate       float64 `perflib:"Retransmission Rate"`
+	TCPReceivedRate          float64 `perflib:"TCP Received Rate"`
+	TCPSentRate              float64 `perflib:"TCP Sent Rate"`
+	TotalReceivedRate        float64 `perflib:"Total Received Rate"`
+	TotalSentRate            float64 `perflib:"Total Sent Rate"`
+	TotalReceivedBytes       float64 `perflib:"Total Received Bytes"`
+	TotalSentBytes           float64 `perflib:"Total Sent Bytes"`
+	UDPPacketsReceivedPersec float64 `perflib:"UDP Packets Received/sec"`
+	UDPPacketsSentPersec     float64 `perflib:"UDP Packets Sent/sec"`
+	UDPReceivedRate          float64 `perflib:"UDP Received Rate"`
+	UDPSentRate              float64 `perflib:"UDP Sent Rate"`
+}
+
+func (c *RemoteFxCollector) collectRemoteFXNetworkCount(ctx *ScrapeContext, ch chan<- prometheus.Metric) (*prometheus.Desc, error) {
+	dst := make([]perflibRemoteFxNetwork, 0)
+	err := unmarshalObject(ctx.perfObjects["RemoteFX Network"], &dst)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, d := range dst {
+		ch <- prometheus.MustNewConstMetric(
+			c.BaseTCPRTT,
+			prometheus.GaugeValue,
+			float64(d.BaseTCPRTT),
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.BaseUDPRTT,
+			prometheus.GaugeValue,
+			float64(d.BaseUDPRTT),
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.CurrentTCPBandwidth,
+			prometheus.GaugeValue,
+			float64(d.CurrentTCPBandwidth),
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.CurrentTCPRTT,
+			prometheus.GaugeValue,
+			float64(d.CurrentTCPRTT),
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.CurrentUDPBandwidth,
+			prometheus.GaugeValue,
+			float64(d.CurrentUDPBandwidth),
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.CurrentUDPRTT,
+			prometheus.GaugeValue,
+			float64(d.CurrentUDPRTT),
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.FECRate,
+			prometheus.GaugeValue,
+			float64(d.FECRate),
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.LossRate,
+			prometheus.GaugeValue,
+			float64(d.LossRate),
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.RetransmissionRate,
+			prometheus.GaugeValue,
+			float64(d.RetransmissionRate),
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.TCPReceivedRate,
```

Comment:
I've run the exporter on my local machine, and a number of the metrics are monotonically increasing, and should be changed to `CounterValue` rather than the current `GaugeValue`.
I noted that this was the case for `net_total_received_rate`, `net_total_sent_rate`, `tcp_sent_rate`, `tcp_received_rate`, `net_total_sent_bytes` and `net_total_received_bytes`. There's likely more metrics than need to be changed.

Reply:
I need to investigate this further but when i checked not all counters are increasing. (exceptions are `services` and `console`  sessions. Since these 2 are not actually remote sessions I have added filters for them. )